core/sparse_array.cpp

```diff
@@ -95,6 +141,31 @@ void SparseArray::Reset(unsigned int p_nrows, unsigned int p_ncols)
 	finished_ = false;
 }
 
+void SparseArray::Resize(uint32_t p_row)
+{
+    //vectors are reallocated by 1000 entries at a time. It can be wasteful but probably better than making a function call for each new entry. Will defer the final decision to Ben. 
```

Comment:
Doubling with each realloc as Peter suggested is usually a good strategy.  I think that's what most C++ standard library implementations do too.  If you start with an initial size of zero, as I suggest above, then the initial width would be enforced here, with something like `new_width = (current_width == 0) ? initial_width : current_width*2;`.

Reply:
Done!